lib/Elevate/Blockers/EA4.pm

```diff
@@ -41,27 +44,73 @@ sub _blocker_ea4_profile ($self) {
 
     Elevate::EA4::backup();
 
+    my @incompatible_packages = $self->_get_incompatible_packages();
+
+    return unless @incompatible_packages;
+
+    my $list = join( "\n", map { "- $_" } @incompatible_packages );
+
+    return $self->has_blocker( <<~"EOS" );
+    One or more EasyApache 4 package(s) are not compatible with $pretty_distro_name.
+    Please remove these packages before continuing the update.
+    $list
+    EOS
+}
+
+sub _get_incompatible_packages ($self) {
+
     my $stash        = Elevate::StageFile::read_stage_file();
     my $dropped_pkgs = $stash->{'ea4'}->{'dropped_pkgs'} // {};
     return unless scalar keys $dropped_pkgs->%*;
 
-    my @incompatible_packages;
-    foreach my $pkg ( sort keys $dropped_pkgs->%* ) {
+    my @incompatible;
+    foreach my $pkg ( sort keys %$dropped_pkgs ) {
         my $type = $dropped_pkgs->{$pkg} // '';
         next if $type eq 'exp';                          # use of experimental packages is a non blocker
         next if $pkg =~ m/^ea-openssl(?:11)?-devel$/;    # ignore these packages, as they can be orphans
-        push @incompatible_packages, $pkg;
+
+        if ( $pkg =~ m/^(ea-php[0-9]+)/ ) {
+            my $php_pkg = $1;
+            next unless $self->_php_version_is_in_use($php_pkg);
+        }
+
+        push @incompatible, $pkg;
     }
 
-    return unless @incompatible_packages;
+    return @incompatible;
+}
 
-    my $list = join( "\n", map { "- $_" } @incompatible_packages );
+sub _php_version_is_in_use ( $self, $php ) {
+    my $current_php_usage = $self->_get_php_versions_in_use();
+    return $current_php_usage->{$php} ? 1 : 0;
```

Comment:
We need to return true if `api_fail` is set in order to assume it is in use so we can block as according to the warning we give if the API fails.